docs: add updated documentation on local registries (#681)
tilt-dev/tilt.build: docs/choosing_clusters.md

A local registry is often the fastest way to speed up your dev experience.

Every cluster sets up this registry slightly differently.

> Tilt-team is currently collaborating with the Kubernetes community on how

Author: done

> protocols for discovery, so that local development tools will auto-configure

Reviewer: idk if necessary, but might read better.
```suggestion
protocols for discovery, so that local development tools like Tilt will auto-configure
```

Author: done

when a local registry is present.

> Tilt currently supports two generic protocols. We've documented both below.

Author: done

> The Kubernetes standard protocol is a KEP that has been vetted by the Kubernetes community.

Reviewer: ```suggestion
The Kubernetes standard protocol is a [KEP](https://github.com/kubernetes/enhancements/tree/master/keps#kubernetes-enhancement-proposals-keps) that has been vetted by the Kubernetes community.
```

Author: done

The annotation-based protocol is used in legacy Tilt scripts.

#### Kubernetes Standard Registry Discovery

> The new protocol uses configmaps in the `kube-public` namespace of your cluster.

Reviewer: what new protocol? either "This protocol uses..." or "This is a new protocol that uses..."

Author: done

If you have a local registry running at `localhost:5000`, apply the following
config map to your cluster:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: local-registry-hosting
  namespace: kube-public
data:
  localRegistryHosting.v1: |
    host: "localhost:5000"
```

> Tilt will now push and pull images from that registry.

Reviewer: ```suggestion
Tilt will automatically detect your local registry, and will push and pull images from it.
```

Author: done
